overlord/devicestate/firstboot.go

```diff
@@ -270,11 +184,7 @@ func populateStateFromSeedImpl(st *state.State, tm timings.Measurer) ([]*state.T
 
 	// now add/chain the tasksets in the right order, note that we
 	// only have tasksets that we did not already seeded
-	sort.Stable(snap.ByType(infos))
-	for _, info := range infos {
-		ts := infoToTs[info]
-		tsAll = chainTs(tsAll, ts)
-	}
+	chainSorted(infos, infoToTs)
 
 	if len(tsAll) == 0 {
 		return nil, fmt.Errorf("cannot proceed, no snaps to seed")
```

Comment:
Not strictly related to the changes here, but I wonder if this could be checked much earlier, above.

Reply:
I want to simplify this code a little bit still in a follow up, I will look at this then